exchanges/order/order_types.go

```diff
@@ -350,28 +350,34 @@ const (
 // Type enforces a standard for order types across the code base
 type Type uint32
 
+// Is checks to see if the enum contains the flag
+func (t Type) Is(in Type) bool {
+	return in != 0 && t&in == in
+}
+
 // Defined package order types
 const (
 	UnknownType Type = 0
 	Limit       Type = 1 << iota
 	Market
 	Stop
-	StopLimit
-	StopMarket
 	TakeProfit
-	TakeProfitMarket
 	TrailingStop
 	IOS
 	AnyType
 	Liquidation
 	Trigger
-	OptimalLimitIOC
-	OCO                              // One-cancels-the-other order
-	ConditionalStop                  // One-way stop order
-	MarketMakerProtection            // market-maker-protection used with portfolio margin mode. See https://www.okx.com/docs-v5/en/#order-book-trading-trade-post-place-order
-	MarketMakerProtectionAndPostOnly // market-maker-protection and post-only mode. Used in Okx exchange orders.
-	TWAP                             // time-weighted average price.
-	Chase                            // chase order. See https://www.okx.com/docs-v5/en/#order-book-trading-algo-trading-post-place-algo-order
+	OCO             // One-cancels-the-other order
+	ConditionalStop // One-way stop order
```

Comment:
Suggestion: This is a bit more work but while you are here, In our order types we can start splitting all these out into specific fields for clarity and trading experience. 

```go
type Submit struct {
    Type        OrderType     // Core order type (e.g., Limit, Market)
    TimeInForce TimeInForce   // Time-in-force (e.g., GTC, IOC, FOK, PostOnly)
    Condition   ConditionType // Conditional trigger (e.g., Stop, TakeProfit)
    SpecialMode SpecialMode   // Exchange-specific modes (e.g., TWAP, MarketMakerProtection)
    // Existing fields (e.g., Pair, Amount, Side, etc.) remain unchanged
}
```

Then you don't need to define hybrid order types.

Reply:
Yeah that is a valid point on causing confusion in the wrapper implementations and increased toil there, I was more coming from a trading experience and clarity on the user side. It's still optional we can circle back to it if we need to change happy to keep as is.